### ISSUE TITLE
Why did glib2 slip through immediately to Slowroll?

### ISSUE THREAD

**Issue by Esokrates** [bmwiedemann/slowroll-tools#5]
I have been hit by https://gitlab.gnome.org/GNOME/glib/-/issues/3105 (regression of glib 2.78.0 that causes crashes in various applications such as network manager) recently and noticed that

https://cdn.opensuse.org/repositories/openSUSE:/ALP:/Experimental:/Slowroll/standard/x86_64/

contains the affected version since Sept. 21, just a day after it landed in Factory:

![image](https://github.com/bmwiedemann/slowroll-tools/assets/17780185/d3b6d1d8-07bd-45d4-9a64-807c02b688f9)

I did not find any notices regarding cve's so I'm wondering if an issue like that could be avoided with slowroll in the future?

**Comment by bmwiedemann** [comment]
You are right that it should not have been picked into Slowroll.

looking at my logs, I see
```
out/log/select-2023-09-21T08:40:23+00:00:upstream patchlevel update in glib2 0
out/log/select-2023-09-21T08:40:23+00:00:submit glib2 now after 622080 s delay

release-2023-09-22T05:24:59+00:00:compare tmp/slo/glib2-2.76.5-1.1.src.rpm tmp/slos.s/glib2-2.78.0-1.1.src.rpm
release-2023-09-22T05:24:59+00:00:osc release openSUSE:ALP:Experimental:Slowroll:Staging --target-project=openSUSE:ALP:Experimental:Slowroll glib2 --target-repository=standard -r standard
```

but the diffs considered were only
```
-rw-r--r-- 1 slowrollbot users 565 Aug  1 14:12 cache/changelogdiff/glib2-2.76.4-1.1-2.76.3-1.2
-rw-r--r-- 1 slowrollbot users 721 Sep 10 06:35 cache/changelogdiff/glib2-2.76.5-1.1-2.76.4-1.2
```

So it seems, I hit a race there. The tools saw the released Tumbleweed version of glib2-2.76.5 that was considered ready after 7 days wait, but Factory had already got 2.78.0 which is what was copied over to the Staging area and released on 2023-09-22.

I could probably use the Tumbleweed version's disturl value to ensure that I get the correct version. That is
```bash
rpm -qp --qf "%{disturl}\n" /ipfs/bafybeicfmqgodmwbpyw2d7w57c2jw2k3sgws5t3oxub2kertroknsrhj4e/history/20230917/tumbleweed/repo/oss/x86_64/glib2-devel-2.76.5-1.1.x86_64.rpm 
obs://build.opensuse.org/openSUSE:Factory/standard/70f31076cbe68a856b4bf87f6d5f3bab-glib2
osc copypac -e openSUSE:Factory -r 70f31076cbe68a856b4bf87f6d5f3bab glib2 openSUSE:ALP:Experimental:Slowroll:Staging
```

**Comment by bmwiedemann** [comment]
As an additional prevention, I have now reduced the timespan between when a new Tumbleweed snapshot is published and when packages are picked up into Slowroll's build project - around 20 minutes.
